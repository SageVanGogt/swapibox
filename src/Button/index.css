.favorites-button, .planets-button, .people-button, .vehicles-button {
  background-color: transparent;
  border: none;
  color: white; }

.favorites-button:active, .planets-button:active, .people-button:active, .vehicles-button:active {
  background-color: chartreuse; }

.favorites-button {
  background-color: black;
  border: none;
  color: white;
  padding: 40px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 4px 2px;
  border-radius: 50%; }

.planets-button, .people-button, .vehicles-button {
  background-color: black;
  border: none;
  color: white;
  padding: 40px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 4px 2px;
  border-radius: 50%; }

.favorite-active {
  background-color: blue; }
