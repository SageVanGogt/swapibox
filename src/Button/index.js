import React from 'react';
// import './index.css';

const Button = (props) => {
  const { section, handleClickEvent } = props
  return (
    <div className='button'>
      <button name={section} onClick={(event) => handleClickEvent(event)}>
        {section}
      </button>
    </div>
  );
}

export default Button;