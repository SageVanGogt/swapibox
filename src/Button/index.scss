.favorites-button, .planets-button, .people-button, .vehicles-button {
  border: 3px solid #ffd54e;
  color: #ffd54e;
  background-color: #3b444b;
  margin: 0px 30px;
  padding: 10px 20px;
  border-radius: 5px;
  font-family: 'Righteous', 'cursive';  
}

.favorites-button:active, .planets-button:active, .people-button:active, .vehicles-button:active {
  background-color: chartreuse;
}

.favorites-button {
  width: 130px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
}

.planets-button, .people-button, .vehicles-button {
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
}

.favorite-active {
  background-color: red;
}

.favorite-counter {
  color: red;
}