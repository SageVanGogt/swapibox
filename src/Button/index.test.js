import React from 'react';
import Button from './index';
import { shallow } from 'enzyme';
import './../setupTests';

describe('Button component', () => {
  it('should render a button', () => {
    const actual = shallow(<Button />)

    expect(actual).toMatchSnapshot();
  });

  it('should render the apppropriate button depending on which section prop is passed in', () => {
    const actual = shallow(<Button section={'vehicles'}/>)

    expect(actual).toMatchSnapshot();
  });

  it('should call handleClickEvent onClick', () => {
    const handleClickEvent = jest.fn();
    const mockEvent = {target: {name: 'vehicles'}};
    const wrapper = shallow(<Button 
      section={'vehicles'} 
      handleClickEvent={handleClickEvent}
    />);

    wrapper.find('button').simulate('click', mockEvent);
    expect(handleClickEvent).toBeCalledWith(mockEvent);
  });
})