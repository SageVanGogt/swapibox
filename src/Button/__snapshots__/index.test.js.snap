// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render a button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        name={undefined}
        onClick={[Function]}
/>,
      "className": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "name": undefined,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          name={undefined}
          onClick={[Function]}
/>,
        "className": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "name": undefined,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button component should render the apppropriate button depending on which section prop is passed in 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    section="vehicles"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        name="vehicles"
        onClick={[Function]}
>
        vehicles
</button>,
      "className": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "vehicles",
        "name": "vehicles",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "vehicles",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          name="vehicles"
          onClick={[Function]}
>
          vehicles
</button>,
        "className": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "vehicles",
          "name": "vehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "vehicles",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
